File: study-planner/scripts/data_fixes/extract_gfr.js
const fs = require('fs');
const { PDFParse } = require('pdf-parse');

async function extract() {
    const buffer = fs.readFileSync('D:\\IP 2026\\13. GFR_2017.pdf');
    const parser = new PDFParse({ data: buffer });
    await parser.load();
    const info = parser.getInfo();
    console.log('Info:', JSON.stringify(info, null, 2));
    
    let allText = '';
    const numPages = info.numPages || info.pages || 100;
    console.log('Attempting to extract', numPages, 'pages...');
    
    for (let i = 1; i <= numPages; i++) {
        try {
            const pageText = await parser.getPageText(i);
            allText += `\n=== PAGE ${i} ===\n` + pageText;
        } catch (e) {
            console.log(`Page ${i} error:`, e.message);
            break;
        }
    }
    
    fs.writeFileSync('D:\\IP 2026\\study-planner\\gfr_2017_extracted.txt', allText);
    console.log('Text length:', allText.length);
    console.log('Done!');
}
extract().catch(console.error);

File: study-planner/scripts/data_fixes/extract_gfr.py
import subprocess
import sys

# Install pypdf if needed
subprocess.check_call([sys.executable, '-m', 'pip', 'install', 'pypdf', '-q'])

from pypdf import PdfReader

pdf_path = r'D:\IP 2026\13. GFR_2017.pdf'
output_path = r'D:\IP 2026\study-planner\gfr_2017_extracted.txt'

reader = PdfReader(pdf_path)
print(f'Number of pages: {len(reader.pages)}')

text = ""
for i, page in enumerate(reader.pages):
    page_text = page.extract_text()
    text += f"\n=== PAGE {i+1} ===\n" + (page_text or '') + "\n"
    if (i+1) % 20 == 0:
        print(f'Extracted {i+1} pages...')

with open(output_path, 'w', encoding='utf-8') as f:
    f.write(text)

print(f'Total text length: {len(text)}')
print('Done!')
